### PR TITLE
Fix test PostAsync_ThrowFromContentCopy_RequestFails

### LINKED ISSUE #26793: [Test failures] System.Net.Http.Functional.Tests.ManagedHandler_HttpClientHandlerTest / PostAsync_ThrowFromContentCopy_RequestFails
It seems that the managed handler throws a different exception than the other ones. I'm looking into it. I'm going to submit a PR to deal with the test failures, but, want to confirm the desired behavior here.

The exception throw by the managed handler (HttpRequestException) seems more appropriate then the one throw by the other handlers, and it has the proper InnerException (the one throw directly by the other handlers). Anyway, let's confirm that this is the desired behavior.

/cc @geoffkizer @stephentoub @karelz

### PR DESCRIPTION
**Pull request by pjanotti**
Fixes #26793
This fixes the test issues mentioned in #26793

### CODE REVIEW

**Comment by pjanotti** [comment]
I still have to test this PR in more platforms so the changes are based on test pass results: 

https://mc.dot.net/#/user/pjanotti/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/94a5109a831fe018e29f3c6ab1354352edd120b7/workItem/System.Net.Http.Functional.Tests

**Comment by pjanotti** [comment]
Affected test is in Outerloop (although it is local) running outer loop:

@dotnet-bot test Outerloop Linux x64 Debug Build 
@dotnet-bot test Outerloop Windows x64 Debug Build 
@dotnet-bot test Outerloop OSX x64 Debug Build 
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build 
@dotnet-bot Test Outerloop NETFX x86 Release Build

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 1]
```diff
@@ -1703,7 +1703,16 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
                         positionGetFunc: () => 0,
                         canReadFunc: () => true,
                         readAsyncFunc: (buffer, offset, count, cancellationToken) => syncFailure ? throw error : Task.Delay(1).ContinueWith<int>(_ => throw error)));
-                    Assert.Same(error, await Assert.ThrowsAsync<FormatException>(() => client.PostAsync(uri, content)));
+
+                    if (!UseManagedHandler && !PlatformDetection.IsUap)
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Nit: I have trouble with the double-negative and would prefer if this were inverted, e.g.
```C#
if (UseManagedHandler || IsUapHandler)
{
    HttpRequestException ...;
    Assert.Same(...);
}
else
{
    Assert.Same(...);
}
```

Comment (karelz) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: How did the test pass on UWP prior to your change? Or was it always failing?

Comment (stephentoub) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: The test is outerloop.  I didn't run outerloop tests on UAP.

Comment (pjanotti) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: This tests are in outer loop so my assumption is that Steve ran outer loop locally but not for UWP. 

As for the managed handler: at the time of his local run it was passing with the original exception but the exception changed after integration with other recent changes to ManagedHandler.

**Comment by pjanotti** [comment]
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot Test Outerloop NETFX x86 Release Build

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 1 above)
Comment (karelz) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: OK, that makes sense - we probably didn't look recently on UWP nightly runs where the failure would have showed up as well.

**Comment by karelz** [comment]
@pjanotti FYI: I updated the top text with "Fixes #26793" to use auto-closing feature.

**Comment by pjanotti** [comment]
CI is taking very long... so far no failures related to the change, restarting some of the pending runs.

@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test UWP CoreCLR x64 Debug Build
@dotnet-bot test Windows x64 Debug Build

**Comment by pjanotti** [comment]
Mergins since failures are not related to this change, issue for failures is #26803F

**Pull request merged**